src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2687,6 +2687,10 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                         Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, distinguisher.First, distinguisher.Second);
                     }
                 }
+                else if (sourceType.SpecialType == SpecialType.System_Void)
+                {
+                    Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, sourceType.ToDisplayString(), targetType.ToDisplayString());
```

Comment:
> It is intentional that this new case for reporting ERR_NoImplicitConv uses the same eager formatting and formatting options as the other cases reported in the binder.

I did a search in the codebase for existing places where we report ERR_NoImplicitConv error. It looks like **none** of them use eager formatting: 
```
src\Compilers\CSharp\Portable\Binder\Binder_Invocation.cs(771):                            Error(diagnostics, ErrorCode.ERR_NoImplicitConv, call.ReceiverOpt.Syntax, distinguisher.First, distinguisher.Second);
src\Compilers\CSharp\Portable\Binder\Binder_Statements.cs(2670):                        Error(diagnostics, ErrorCode.ERR_NoImplicitConvCast, syntax, distinguisher.First, distinguisher.Second);
src\Compilers\CSharp\Portable\Binder\Binder_Statements.cs(2687):                        Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, distinguisher.First, distinguisher.Second);
src\Compilers\CSharp\Portable\Binder\Binder_Statements.cs(2693):                    Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, distinguisher.First, distinguisher.Second);
```

Please follow the guidelines for avoiding premature symbol formatting when a diagnostics is created. I see nothing special about this case and we intentionally added an infrastructure that makes doing this very easy, even when a custom formatting is needed (see the usages of FormattedSymbol).